service/src/test/java/bio/terra/workspace/service/workspace/WorkspaceServiceTest.java

```diff
@@ -481,6 +515,86 @@ void createGoogleContextRawlsStageThrows() throws Exception {
+            .cloningInstructions(CloningInstructions.COPY_RESOURCE)
+            .resourceId(UUID.randomUUID())
+            .workspaceId(sourceWorkspaceId)
+            .managedBy(ManagedByType.MANAGED_BY_USER)
+            .privateResourceState(PrivateResourceState.INITIALIZING)
+            .accessScope(AccessScopeType.ACCESS_SCOPE_PRIVATE)
+            .applicationId(null)
+            .iamRole(ControlledResourceIamRole.OWNER)
+            .assignedUser(USER_REQUEST.getEmail())
+            .build())
+        .build();
+    ApiGcpGcsBucketCreationParameters creationParameters = new ApiGcpGcsBucketCreationParameters()
+        .name("foo")
+        .defaultStorageClass(ApiGcpGcsBucketDefaultStorageClass.NEARLINE)
+            .lifecycle(new ApiGcpGcsBucketLifecycle()
+                .addRulesItem(new ApiGcpGcsBucketLifecycleRule()
+                    .condition(new ApiGcpGcsBucketLifecycleRuleCondition()
+                        .age(90))
+                    .action(new ApiGcpGcsBucketLifecycleRuleAction()
+                        .type(ApiGcpGcsBucketLifecycleRuleActionType.SET_STORAGE_CLASS)
+                        .storageClass(ApiGcpGcsBucketDefaultStorageClass.STANDARD))));
+
+    controlledResourceService.createControlledResourceSync(
+        bucketResource,
+        ControlledResourceIamRole.OWNER,
+        USER_REQUEST,
+        creationParameters);
+
+    final Workspace destinationWorkspace = defaultRequestBuilder(UUID.randomUUID())
+        .displayName("Destination Workspace")
+        .description("Copied from source")
+        .spendProfileId(new SpendProfileId(SPEND_PROFILE_ID))
+        .build();
+    final String destinationLocation = "us-east1";
+    final String cloneJobId = workspaceService.cloneWorkspace(
+        sourceWorkspaceId, USER_REQUEST, destinationLocation, destinationWorkspace);
+    jobService.waitForJob(cloneJobId);
+    JobResultOrException<ApiClonedWorkspace> cloneResultOrException = jobService.retrieveJobResult(cloneJobId, ApiClonedWorkspace.class, USER_REQUEST);
+    assertNull(cloneResultOrException.getException());
+    ApiClonedWorkspace cloneResult = cloneResultOrException.getResult();
+    assertEquals(destinationWorkspace.getWorkspaceId(), cloneResult.getDestinationWorkspaceId());
+    assertThat(cloneResult.getResources(), hasSize(1));
+
+    // destination WS should exist
+    final Workspace retrievedDestinationWorkspace = workspaceService.getWorkspace(
+        destinationWorkspace.getWorkspaceId(), USER_REQUEST);
+    assertEquals("Destination Workspace", retrievedDestinationWorkspace.getDisplayName().orElseThrow());
+    assertEquals("Copied from source", retrievedDestinationWorkspace.getDescription().orElseThrow());
+    assertEquals(WorkspaceStage.MC_WORKSPACE, retrievedDestinationWorkspace.getWorkspaceStage());
+
+    // Destination Workspace should have a GCP context
+    assertNotNull(gcpCloudContextService.getGcpCloudContext(destinationWorkspace.getWorkspaceId()).orElseThrow());
+
```

Comment:
It looks like you're also cloning a bucket from the source workspace, should this have any assertions about the cloned bucket?